Kernel/Net/NetworkAdapter.cpp

```diff
@@ -122,7 +122,7 @@ RefPtr<PacketWithTimestamp> NetworkAdapter::acquire_packet_buffer(size_t size)
         if (unused_packet->buffer->capacity() >= size)
             return unused_packet;
 
-        // FIXME: Shouldn't we put existing buffers that are too small back into the list?
+        unused_packets.append(*unused_packet);
```

Comment:
Since we only pop the first packet every time, does this create a situation where a tiny packet could sit at the head of the queue and prevent anyone who doesn't need a tiny packet from using an otherwise full queue?

Reply:
No, as we are appending, not prepending. It would get popped once, determined as being to small, and put back in at the end of the queue.